database/migrations/2024_08_23_165504_create_course_table.php

```diff
@@ -17,11 +18,11 @@ public function up()
             $table->id();  // Primary key, auto-incrementing
             $table->string('code', 16)->unique();  // Course code with a unique constraint
             $table->foreignId('semester_id')->constrained('semesters');  // Foreign key to semesters.id
-            $table->enum('academic_program', ['Undergraduate', 'Postgraduate']);  // Enum for academic program
-            $table->integer('version');  // Version for curriculum
+            $table->enum('academic_program', array_values(Course::getAcademicPrograms()));  // Enum for academic program 
+            $table->enum('version', array_keys(Course::getVersions()));  // Enum for version as numeric keys
```

Comment:
Let's keep this as an integer in the db table since we might introduce new versions later (If we use Enum, adding a new version might be difficult).